kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/kommunicate/utils/KmThemeHelper.java

```diff
@@ -0,0 +1,110 @@
+package com.applozic.mobicomkit.uiwidgets.kommunicate.utils;
+
+import android.content.Context;
+import android.graphics.Color;
+import android.text.TextUtils;
+
+import com.applozic.mobicomkit.uiwidgets.AlCustomizationSettings;
+import com.applozic.mobicomkit.uiwidgets.R;
+import com.applozic.mobicommons.ApplozicService;
+
+import io.kommunicate.callbacks.KmCallback;
+import io.kommunicate.utils.KmAppSettingPreferences;
+
+public class KmThemeHelper implements KmCallback {
+
+    private static KmThemeHelper kmThemeHelper;
+    private Context context;
+    private KmAppSettingPreferences appSettingPreferences;
+    private AlCustomizationSettings alCustomizationSettings;
+    private int primaryColor = -1;
+    private int secondaryColor = -1;
+    private int sentMessageBackgroundColor = -1;
+    private int sendButtonBackgroundColor = -1;
+    private int sentMessageBorderColor = -1;
+
+    public static KmThemeHelper getInstance(Context context, AlCustomizationSettings alCustomizationSettings) {
+        if (kmThemeHelper == null) {
+            kmThemeHelper = new KmThemeHelper(context, alCustomizationSettings);
+        }
+        return kmThemeHelper;
+    }
+
+    private KmThemeHelper(Context context, AlCustomizationSettings alCustomizationSettings) {
+        this.context = ApplozicService.getContext(context);
+        this.alCustomizationSettings = alCustomizationSettings;
+        appSettingPreferences = KmAppSettingPreferences.getInstance();
+        appSettingPreferences.setCallback(this);
+    }
+
+    public int getPrimaryColor() {
+        if (primaryColor == -1) {
+            String colorStr = appSettingPreferences.getPrimaryColor();
+            primaryColor = !TextUtils.isEmpty(colorStr) ? Color.parseColor(colorStr) : context.getResources().getColor(R.color.applozic_theme_color_primary);
+        }
+        return primaryColor;
+    }
+
+    public int getSentMessageBackgroundColor() {
```

Comment:
@reytum this and below 2 methods code looks similar, is it possible to find a way to create a single method that can work for all the 3?

Reply:
They look same because we are setting only one color for all. If customer wants to set different color then different methods are required. Seperate methods are there because these colors are different in applozic-settings.json file. This is for future, if we provide option to set different colors.